cwl/src/main/scala/cwl/CommandLineTool.scala

```diff
@@ -85,14 +87,68 @@ case class CommandLineTool private(
     requirement.fold(RequirementToAttributeMap).apply(inputNames)
   }
 
+  /*
+   * The command template is built following the rules described here: http://www.commonwl.org/v1.0/CommandLineTool.html#Input_binding
+   * - The baseCommand goes first
+   * - Then the arguments are assigned a sorting key and transformed into a CommandPart
+   * - Finally the inputs are folded one by one into a CommandPartsList
+   * - arguments and inputs CommandParts are sorted according to their sort key
+   */
+  private [cwl] def buildCommandTemplate(inputValues: WomEvaluatedCallInputs): ErrorOr[List[CommandPart]] = {
+    import cats.instances.list._
+    import cats.syntax.traverse._
+    import common.validation.Validation._
+
+    val baseCommandPart = baseCommand.toList.flatMap(_.fold(BaseCommandToCommandParts))
+
+    val argumentsParts: CommandPartsList =
+    // arguments is an Option[Array[Argument]], the toList.flatten gives a List[Argument]
+      arguments.toList.flatten
+        // zip the index because we need it in the sorting key
+        .zipWithIndex.foldLeft(CommandPartsList.empty)({
+        case (commandPartsList, (argument, index)) =>
+          val part = argument.fold(ArgumentToCommandPart)
+          // Get the position from the binding if there is one
+          val position = argument.select[ArgumentCommandLineBinding].flatMap(_.position)
+            .map(Coproduct[StringOrInt](_)).getOrElse(DefaultPosition)
+
+          // The key consists of the position followed by the index
+          val sortingKey = CommandBindingSortingKey(List(position, Coproduct[StringOrInt](index)))
+
+          commandPartsList :+ SortKeyAndCommandPart(sortingKey, part)
+      })
+
+    val inputBindingsCommandParts: ErrorOr[List[SortKeyAndCommandPart]] = inputs.toList.flatTraverse[ErrorOr, SortKeyAndCommandPart]({
+      inputParameter =>
+        val parsedName = FullyQualifiedName(inputParameter.id)(ParentName.empty).id
+        val womType = inputParameter.`type`.map(_.fold(MyriadInputTypeToWomType)).getOrElse(WomStringType)
+        val defaultValue = inputParameter.default map {
+          case CwlAny.Json(jsonDefault) => womType.coerceRawValue(jsonDefault.sprayJsonRepresentation).get
+          case CwlAny.File(fileDefault) => womType.coerceRawValue(fileDefault.path.get).get
+        }
+
+        inputValues
+          .find(_._1.name == parsedName).map(_._2)
```

Comment:
ToL I'm personally not a fan of `_1` and `_2`, prefer `{ case (something, _) => }` (even though it is more verbose).